packages/server-wallet/src/utilities/signatures.ts

```diff
@@ -1,59 +1,25 @@
-import {instantiateSecp256k1, Secp256k1, RecoveryId} from '@bitauth/libauth';
-import {utils, Wallet} from 'ethers';
-import {State, StateWithHash} from '@statechannels/wallet-core';
-
-let secp256k1: Secp256k1;
-export const initialized: Promise<any> = instantiateSecp256k1().then(m => (secp256k1 = m));
+import {Wallet} from 'ethers';
+import {State, toNitroState} from '@statechannels/wallet-core';
+import * as wasmUtils from '@statechannels/wasm-utils';
+import {State as NitroState} from '@statechannels/nitro-protocol';
 
 const knownWallets: Record<string, string> = {};
 const cachedAddress = (privateKey: string): string =>
   knownWallets[privateKey] || (knownWallets[privateKey] = new Wallet(privateKey).address);
 
-export async function fastSignState(
-  state: StateWithHash,
+export async function signState(
+  state: State,
   privateKey: string
 ): Promise<{state: State; signature: string}> {
   const address = cachedAddress(privateKey);
   if (state.participants.map(p => p.signingAddress).indexOf(address) < 0) {
     throw new Error("The state must be signed with a participant's private key");
   }
 
-  const {stateHash} = state;
-
-  const signature = await fastSignData(stateHash, privateKey);
+  const signature = await wasmUtils.signState(toNitroState(state), privateKey).signature;
```

Comment:
I think this is now synchronous.

With bitauth, we needed to await the EC to be constructed, and thus this needed to be async.

But the wasm-utils are synchronous -- I think it probably constructs a EC on each call, which is suboptimal, but it still "fast enough".

Reply:
#2619